src/TSTransformer/nodes/class/transformDecorators.ts

```diff
@@ -55,7 +60,7 @@ function transformMemberDecorators(
 
 		luau.list.pushList(initializers, prereqs);
 
-		if (!canInline && !luau.isSimple(expression)) {
+		if (!canInline && !expressionMightMutate(state, expression, decorator.expression)) {
```

Comment:
I think it's confusing that `canInline` doesn't encompass _all_ the inline conditions. Also, will this unnecessarily prereq the last decorator, if there's multiple? With the reverse order of execution, I don't think that's necessary.

Reply:
I agree, will iterate on this a bit more. Two problems I need to figure out:

1. `canInline` is looking at all of the decorators and not a specific decorator.
2. We can't inline the last decorator if it's a either:
    - method decorator and which has parameter decorators following it
    - class decorator and which has constructor parameter decorators following it